src/File.ts

```diff
@@ -178,20 +206,28 @@ Recommendations:
             // so write out to the console a representation of the data needed to reconstruct the above
             // findLineNumberOfTaskToToggle() call, so that the content can be saved
             // to a JSON file and then re-used in a 'unit' test.
-            saveMockDataForTesting(originalTask, fileLines, listItemsCache);
+            saveMockDataForTesting(task, fileLines, listItemsCache);
         }
-        return retry();
+        throw new WarningWorthRetrying();
```

Comment:
Hi, many thanks for this contribution.

I think that the change from just retry() to warning as well will generate an annoying number of warnings - like 10 - for users under some circumstances.

I added some test files describing reproductions for various failures to find lines recently - please see the files in the following directory, and it would help if you could run through those scenarios in those files and confirm that you feel the new behaviour would be OK to release.

https://github.com/obsidian-tasks-group/obsidian-tasks/tree/main/resources/sample_vaults/Tasks-Demo/Manual%20Testing/Task%20Toggling%20Scenarios

Many thanks.

Reply:
There is no change of behavior here (assuming I did not fail in the refactor). When `WarningWorthRetrying` is thrown without a message it will not show a warning, just like in the original code.
Nice to know about these tests, I'll check them out.

Reply:
> When `WarningWorthRetrying` is thrown without a message it will not show a warning, just like in the original code.

Ah right. Thanks.

I wondered if it might be possible to convey that in the name - although I wasn't able to come up with something in a couple of minutes..

Reply:
Sure, I can make it a separate exception class, that will be the clearest option.

Reply:
Done.